azurerm/internal/services/storage/resource_arm_hpc_cache_nfs_target.go

```diff
@@ -0,0 +1,288 @@
+		Delete: resourceArmHPCCacheNFSTargetDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parsers.HPCCacheTargetID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: storageValidate.HPCCacheTargetName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"cache_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"host_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"usage_model": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"READ_HEAVY_INFREQ",
+					"WRITE_WORKLOAD_15",
+					"WRITE_AROUND",
+				}, false),
+			},
+
+			"namespace_junction": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MinItems: 1,
+				// Confirmed with service team that they have a mac of 10 that is enforced by the backend.
+				MaxItems: 10,
+				Elem:     namespaceJunctionResource(),
+			},
+		},
+	}
+}
+
+func resourceArmHPCCacheNFSTargetCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Storage.StorageTargetsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure HPC Cache NFS Target creation.")
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	cache := d.Get("cache_name").(string)
+
+	if d.IsNewResource() {
+		resp, err := client.Get(ctx, resourceGroup, cache, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Error checking for existing HPC Cache NFS Target %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return tf.ImportAsExistsError("azurerm_hpc_cache_nfs_target", *resp.ID)
+		}
+	}
+
+	hostName := d.Get("host_name").(string)
+	usageModel := d.Get("usage_model").(string)
+	namespaceJunctions := expandNamespaceJunctions(d.Get("namespace_junction").(*schema.Set).List())
+
+	// Construct parameters
+	param := &storagecache.StorageTarget{
+		StorageTargetProperties: &storagecache.StorageTargetProperties{
+			Junctions:  namespaceJunctions,
+			TargetType: storagecache.StorageTargetTypeNfs3,
+			Nfs3: &storagecache.Nfs3Target{
+				Target:     &hostName,
+				UsageModel: &usageModel,
```

Comment:
i personally find it cleaner to just do this where possible
```suggestion
				UsageModel: &utils.String(d.Get("usage_model").(string)),
```

Reply:
Sure, also same for other redundent auto vars.